test/PlaidClientTest.js

```diff
@@ -594,6 +594,114 @@ describe('plaid.Client', () => {
 
     });
 
+    describe('assets', () => {
+      var days_requested = 60;
+      var options = {
+        client_report_id: "reportid123",
+        user: {
+          client_user_id: "userid123",
+          first_name: "first",
+          middle_name: "middle",
+          last_name: "last",
+          ssn: "999-60-1111",
+          phone_number: "(123)456-7890",
+          email: "hello@test.com",
+        },
+      };
+      var auditor_id = "fannie_mae";
+
+      var createAssetReport = (cb) => {
+        pCl.createAssetReport([testAccessToken], days_requested, options,
+        (err, response) => {
+          expect(err).to.be(null);
+          expect(response).to.be.ok();
+          expect(response.request_id).to.be.ok();
+          expect(response.asset_report_token).to.be.ok();
+          expect(response.asset_report_id).to.be.ok();
+
+          cb(null, response.asset_report_token);
+        });
+      }
+
+      var getAssetReport = async (asset_report_token, num_retries_remaining, cb) => {
+        if (num_retries_remaining <= 0) {
+          throw new Error('Ran out of retries while polling for asset report');
+        }
+
+        pCl.getAssetReport(asset_report_token, async (err, response) => {
+          if (err) {
+            if (err.status_code === 400) {
```

Comment:
Let's check the error code as well, so that we only enter this wait if `status_code === 400` and the error code is for asset report not ready.

Reply:
⚡️